### PR TITLE
Make it possible to submit topology in inactive state

### PR DESCRIPTION
**Pull request by jasonjckn**
https://github.com/nathanmarz/storm/issues/267

### CODE REVIEW

File: src/clj/backtype/storm/daemon/nimbus.clj [hunk 1]
```diff
@@ -867,215 +867,226 @@
...
-          (when-not is
-            (throw (RuntimeException.
-                    "Could not find input stream for that id")))
-          (let [ret (.read is)]
-            (.put downloaders id is)
-            (when (empty? ret)
-              (.remove downloaders id))
-            (ByteBuffer/wrap ret)
-            )))
+    (letfn [(submit-topology-with-opts
```
Comment (nathanmarz) [inline src/clj/backtype/storm/daemon/nimbus.clj]: Why is it necessary to have another helper function here? Can't you have submitTopology just call submitTopologyWithOpts directly? Also, I think it's better for helper functions to just be their own functions rather than in the same form of the service handler via letfn.

File: test/clj/backtype/storm/integration_test.clj [hunk 2]
```diff
@@ -276,6 +276,50 @@
...
+      (reset! bolt-prepared? false)
+      (reset! spout-opened? false)
+      
+      (submit-local-topology-with-opts (:nimbus cluster)
+        "test"
+        {}
+        (:topology tracked)
+        (SubmitOptions. TopologyInitialStatus/INACTIVE))
+      (.feed feeder [1])
+      (Thread/sleep 5000)
```
Comment (nathanmarz) [inline test/clj/backtype/storm/integration_test.clj]: Why is this necessary?

**Comment by nathanmarz** [comment]
Looks really good, except for a couple minor comments.

File: test/clj/backtype/storm/integration_test.clj (hunk 2 above)
Comment (jasonjckn) [inline test/clj/backtype/storm/integration_test.clj]: why is what necessary?

File: src/clj/backtype/storm/daemon/executor.clj [hunk 3]
```diff
@@ -343,19 +344,19 @@
     ret
     ))
 
+(defn sleep-until-true [wait-fn]
```
Comment (nathanmarz) [inline src/clj/backtype/storm/daemon/executor.clj]: Isn't it simpler to just use "while" here?

Comment (jasonjckn) [inline src/clj/backtype/storm/daemon/executor.clj]: I learned something new in clojure core.  that hasn't happened for a while :)
